httpApi.js

```diff
@@ -144,13 +180,25 @@ async function getStakers () {
   return res
 }
 
-async function getJobValues (jobId) {
-  let blockNumber = await web3.eth.getBlockNumber()
+async function getJobValues (jobId, provider) {
+  let blockNumber
+  let fulfills
 
-  let fulfills = await jobManager.getPastEvents('JobReported', {
-    fromBlock: Math.max(0, Number(blockNumber) - 1000),
-    toBlock: 'latest'
-  })
+  if (!isInfura(provider)) {
```

Comment:
I think, instead of writing `if....else` on such big blocks, you can define a variable, and assign it the value of eth or geth depending on whether it is using infura as the provider or not and then simply call the contract function once.

Reply:
Yeah got it. Do you mean something like:
```
const infura = isInfura(provider)
let blockNumber = infura ? await web3.eth.getBlockNumber() : await web3Geth.eth.getBlockNumber()
```
If these are the changes required, shall I replicate them in all other function?

Reply:
Rather I would suggest:
```
let web3Provider = web3;
let manager = jobManager;
if(!isInfura(provider)) {
     web3Provider = web3Geth;
     manager = gethJobManager;
}

blockNumber = await web3Provider.eth.getBlockNumber;

fulfills = await manager.getPastEvents('JobReported', {
      fromBlock: Math.max(0, Number(blockNumber) - 1000),
      toBlock: 'latest'
    })
```